submissions/iakow/OOP-exercise/index.js

```diff
@@ -0,0 +1,83 @@
+class Inhabitant {
+   constructor({ species, name, say, gender }) {
+      this.species = species;
+      this.name = name;
+      this.say = say;
+      this.gender = gender;
+      this.friends = new Set();
+   }
+
+   makeFriends(...applicants) {
+      applicants.filter(applicant => applicant instanceof Inhabitant && applicant != this)
+         .forEach(applicant => this.friends.add(applicant));
+   }
+
+   toString() {
+      const { name, species, friends, gender, say } = this;
+
+      const friendNames = [];
+      friends.forEach((friend) => friendNames.push(friend.name));
+      const friendsNamesString = friendNames.length ? `${friendNames.join(', ')};` : '';
+
+      return `${species}; ${name}; ${gender}; ${say}; ${friendsNamesString}`;
```

Comment:
There is a pattern here.
Consider the following example: `return [propA, propB, varC].join("; ");`
The latter is more scalable: you do not need to remember about `; `, no need to deal with extra-long string if number of properties grows.
Eliminate code repetition whenever possible to make this a habit.